pandas/core/frame.py

```diff
@@ -1598,6 +1612,18 @@ def to_records(self, index=True, convert_datetime64=None):
         >>> df.to_records(index=False)
         rec.array([(1, 0.5 ), (2, 0.75)],
                   dtype=[('A', '<i8'), ('B', '<f8')])
+
+        Data types can be specified for the columns:
+
+        >>> df.to_records(column_dtypes={"A": "int32"})
```

Comment:
can you add an example which looks something like the OP, e.g. 

``df.to_records(index_dtypes="<S{}".format(df.index.str.len().max()))``

Reply:
Sure, done.